entity-management-web/src/main/java/eu/europeana/entitymanagement/batch/config/EntityUpdateJobFactory.java

```diff
@@ -0,0 +1,171 @@
+import eu.europeana.entitymanagement.definitions.batch.model.BatchEntityRecord;
+import eu.europeana.entitymanagement.definitions.batch.model.ScheduledTaskType;
+import eu.europeana.entitymanagement.definitions.batch.model.ScheduledUpdateType;
+import org.springframework.batch.core.ItemProcessListener;
+import org.springframework.batch.core.Job;
+import org.springframework.batch.core.Step;
+import org.springframework.batch.core.StepExecutionListener;
+import org.springframework.batch.core.configuration.annotation.JobBuilderFactory;
+import org.springframework.batch.core.configuration.annotation.StepBuilderFactory;
+import org.springframework.batch.core.launch.support.RunIdIncrementer;
+import org.springframework.batch.core.step.skip.SkipPolicy;
+import org.springframework.batch.item.ItemProcessor;
+import org.springframework.batch.item.ItemReader;
+import org.springframework.batch.item.ItemWriter;
+import org.springframework.batch.item.support.CompositeItemProcessor;
+import org.springframework.batch.item.support.CompositeItemWriter;
+import org.springframework.batch.item.support.SynchronizedItemStreamReader;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.task.TaskExecutor;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+import java.util.Arrays;
+import java.util.List;
+
+import static eu.europeana.entitymanagement.batch.utils.BatchUtils.*;
+import static eu.europeana.entitymanagement.common.vocabulary.AppConfigConstants.*;
+
+/**
+ * Entity Job update factory class
+ * @author srishti singh
+ * @since 14 July 2025
+ */
+@Component
+public class EntityUpdateJobFactory {
+
+    private final JobBuilderFactory jobBuilderFactory;
+    private final StepBuilderFactory stepBuilderFactory;
+    private final ScheduledTaskItemListener itemListener;
+    private final ScheduledTaskService scheduledTaskService;
+
+    /** SkipPolicy to ignore all failures when executing jobs, as they can be handled later */
+    private final SkipPolicy noopSkipPolicy = (Throwable t, int skipCount) -> true;
+
+    @Resource
+    EntityManagementConfiguration emConfig;
+
+    @Autowired
+    ApplicationContext applicationContext;
+
+    public EntityUpdateJobFactory(JobBuilderFactory jobBuilderFactory, StepBuilderFactory stepBuilderFactory,
+                                  ScheduledTaskItemListener itemListener,
+                                  ScheduledTaskService scheduledTaskService) {
+        this.jobBuilderFactory = jobBuilderFactory;
+        this.stepBuilderFactory = stepBuilderFactory;
+        this.itemListener = itemListener;
+        this.scheduledTaskService = scheduledTaskService;
+    }
+
+    // TODO this is a execution, we should seperate it from the factory methods and step creations
```

Comment:
please keep the new class clean, to only build the Job object, the execution needs to be called from the update service